pyquil/paulis.py

```diff
@@ -682,6 +682,26 @@ def exp_wrap(param):
     return exp_wrap
 
 
+def exponentiate_pauli_sum(pauli_sum):
```

Comment:
I would prefer calling this `exponentiate_commuting_pauli_sum` as it is otherwise not correct/misleading.

Reply:
Alternatively we can call it `exponentiate_pauli_terms(pauli_terms)` and then have some math in the documentation explain what unitary this corresponds to, i.e.

    exponentiate_pauli_terms([p1, p2]) == exp(1j*p2) * exp(1j*p1)

or something like that.

Reply:
I super-agree with this.

Reply:
:+1:

Reply:
I agree as well. Will make these changes.